gensim/sklearn_api/basemodel.py

```diff
@@ -11,7 +11,7 @@
 from abc import ABCMeta, abstractmethod
 
 
-class BaseSklearnWrapper(object):
+class BaseTransformer(object):
```

Comment:
OK, providing a place to include a error/warning is a potential reason to implement `partial_fit()` even where the underlying algorithm can't (or shouldn't casually) be used that way. BUT, simply not implementing a method is also a pretty good way of signalling that something isn't implemented – and indeed that seems to be what scikit-learn itself does. For that reason, it's even possible other code will use introspection about whether an sklearn pipeline object implements `partial_fit()` as a test of support, in which case having an implemented-method that just errors could be suboptimal. 

But with `set_params()`, again, why have any lesser-featured implementation here, when every subclass (so far) inherits from scikit-learn's own `BaseEstimator`, which provides a more fully-featured implementation? Why wouldn't we want fewer limitations (and match the source library's behavior) for zero lines of code, rather than more limitations with more code?